example/ida/graph_ir.py

```diff
@@ -133,12 +135,14 @@ def Show(self):
 
 print "IR ok... %x" % ad
 
-for irb in ir_arch.blocks.values():
-    for assignblk in irb.irs:
-        for dst, src in assignblk.items():
-            del(assignblk[dst])
+for irb in ir_arch.blocks.itervalues():
+    for i, assignblk in enumerate(irb.irs):
+        new_assignblk = dict(assignblk)
+        for dst, src in assignblk.iteritems():
+            del new_assignblk[dst]
             dst, src = expr_simp(dst), expr_simp(src)
-            assignblk[dst] = src
+            new_assignblk[dst] = src
```

Comment:
Ok, here is remark:
maybe you ripped the pattern from elsewhere, but `new_assignblk` is initialized from the dictionary from the old `assignblk`, but the first action we do in the loop is erasing its keys one by one.
What about creating an empty dictionary, and then adding fresh new keys?